### PR TITLE
Implement Hilt

### PR DESCRIPTION
**Pull request by AntoniKwok**
Hi, I've made a few changes:

- Using Hilt
- Using MVVM Architecture
- Using Diff Util
- Using Coroutines

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -28,6 +30,10 @@ android {
     buildFeatures {
         viewBinding true
     }
+
+    kotlinOptions {
+        jvmTarget = "1.8"
+    }
 }
```
Comment (naufalprakoso) [inline app/build.gradle:43]: Hilt uses Java 8 features, better to enable Java 8 in this project:
```
compileOptions {
    sourceCompatibility JavaVersion.VERSION_1_8
    targetCompatibility JavaVersion.VERSION_1_8
  }
```

File: app/build.gradle [hunk 2]
```diff
@@ -52,4 +58,11 @@ dependencies {
 
     // Material dependencies
     implementation "com.google.android.material:material:$material_version"
+
```
Comment (naufalprakoso) [inline app/build.gradle:67]: Nit: Please add
`// Hilt`

File: app/src/main/java/com/naufalprakoso/billreminder/ui/bill/detail/BillDetailActivity.kt [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
 import android.view.View
+import androidx.appcompat.app.AppCompatActivity
 import com.naufalprakoso.billreminder.database.AppDatabase
 import com.naufalprakoso.billreminder.database.DbWorkerThread
 import com.naufalprakoso.billreminder.database.entity.Bill
 import com.naufalprakoso.billreminder.databinding.ActivityBillDetailBinding
 import com.naufalprakoso.billreminder.utils.BILL_ID
 import java.text.NumberFormat
-import java.util.Locale
+import java.util.*
```
Comment (naufalprakoso) [inline app/src/main/java/com/naufalprakoso/billreminder/ui/bill/detail/BillDetailActivity.kt]: Better to not use wildcard import in here

File: app/src/main/java/com/naufalprakoso/billreminder/ui/main/all/BillAdapter.kt [hunk 4]
```diff
@@ -8,7 +8,7 @@ import androidx.recyclerview.widget.RecyclerView
 import com.naufalprakoso.billreminder.database.entity.Bill
 import com.naufalprakoso.billreminder.databinding.ItemBillUnpaidBinding
 import java.text.NumberFormat
-import java.util.Locale
+import java.util.*
```
Comment (naufalprakoso) [inline app/src/main/java/com/naufalprakoso/billreminder/ui/main/all/BillAdapter.kt]: Better to not use wildcard import in here

**Review by naufalprakoso — CHANGES_REQUESTED** [review]
Thanks for your contribution 🎉. After I test and checked it, I found an issue. After adding a new bill data, the bills did not show in the Bills Unpaid and Bills page.

**Comment by AntoniKwok** [comment]
Hi @naufalprakoso , Thank you for your feedback and comment

File: app/src/main/java/com/naufalprakoso/billreminder/utils/DiffUtilCallback.kt [hunk 5]
```diff
@@ -0,0 +1,31 @@
+package com.naufalprakoso.billreminder.utils
+
+import androidx.annotation.Nullable
+import androidx.recyclerview.widget.DiffUtil
+import com.naufalprakoso.billreminder.database.entity.Bill
+
+class DiffUtilCallback(
```
Comment (naufalprakoso) [inline app/src/main/java/com/naufalprakoso/billreminder/utils/DiffUtilCallback.kt:7]: Nice 🎉

**Review by naufalprakoso — APPROVED** [review]
Thanks for your great contribution 🔥

**Pull request merged**